fw/http.c

```diff
@@ -5347,6 +5388,39 @@ tfw_h2_resp_adjust_fwd(TfwHttpResp *resp)
 	if (unlikely(r))
 		goto clean;
 
+	if (resp->trailers_len > 0) {
+		// MUST _NOT_ use resp pool to encode trailer headers
```

Comment:
as I said in the comment, the pool-related functions assume the body must be after the headers and no trailer headers, so when the body is written to the skb, you have no chance to append new headers between the original headers (be fixed in length then) and the body anymore, if you do, it will overwrite the data between them and cause memory violation, which I have tested and verified. Of course, we can rewrite those functions, but the cost is very high, so why not use simple wrapper code to tackle it?

Reply:
@krizhanovsky what do you think?